src/Sylius/Bundle/ResourceBundle/Controller/Configuration.php

```diff
@@ -316,14 +316,26 @@ public function getRequestParameter($parameter, $defaults = array())
         );
     }
 
-    public function getMethod($default)
+    public function getRepositoryMethod($default)
     {
-        return $this->parameters->get('method', $default);
+        $repository = $this->parameters->get(
+            'repository',
+            array(
+                'method' => $this->parameters->get('method', $default)
+            )
+        );
+
+        return is_array($repository) ? $repository['method'] : $repository;
     }
 
-    public function getArguments(array $default = array())
+    public function getRepositoryArguments(array $default = array())
     {
-        return $this->parameters->get('arguments', $default);
+        $repository = $this->parameters->get(
```

Comment:
@pjedrzejewski I prefer break the BC and do the same thing than the factory. what do you think ?